Source/DafnyCore/Rewriters/InductionRewriter.cs

```diff
@@ -195,6 +213,49 @@ void ComputeInductionVariables<TVarType>(IToken tok, List<TVarType> boundVars, E
     }
   }
 
+  /// <summary>
+  /// Obtain and return matching patterns for
+  ///     (forall inductionVariables :: body)
+  /// If there aren't any, then return null.
+  /// </summary>
+  List<List<Expression>> ComputeInductionTriggers(List<Expression> inductionVariables, Expression body, ModuleDefinition moduleDefinition) {
+    Contract.Requires(inductionVariables.Count != 0);
+
+    // Construct a quantifier, because that's what the trigger-generating machinery expects.
```

Comment:
That would be better. When I started this PR, the resolver only computed the list of variables to use for the quantifier in the induction. The rest is done in the verifier (Boogie generator). I think one reason for that was that only the verifier knew how to construct the "decreases" antecedent. We could change this, since the source language now supports `decreases to` expression. But I suggest we would do such a refactoring in a different PR.